internal/provider/repository_branch_protection_resource.go

```diff
@@ -0,0 +1,816 @@
+				msg = fmt.Sprintf("Branch protection not found for %s/%s/%s: %s", owner, repo, branchName, err)
+			default:
+				msg = fmt.Sprintf("Error getting branch protection: %s", err)
+			}
+		} else {
+			msg = fmt.Sprintf("Failed to get branch protection: %s", err)
+		}
+		response.Diagnostics.AddError("Unable to import branch protection", msg)
+		return
+	}
+
+	// Map response to model
+	var data repositoryBranchProtectionResourceModel
+	data.ID = types.StringValue(request.ID)
+	data.Owner = types.StringValue(owner)
+	data.Repo = types.StringValue(repo)
+	data.BranchName = types.StringValue(branchName)
+
+	// Map protection settings
+	data.EnablePush = types.BoolValue(protection.EnablePush)
+	data.EnablePushWhitelist = types.BoolValue(protection.EnablePushWhitelist)
+	data.PushWhitelistDeployKeys = types.BoolValue(protection.PushWhitelistDeployKeys)
+	data.EnableStatusCheck = types.BoolValue(protection.EnableStatusCheck)
+	data.RequireSignedCommits = types.BoolValue(protection.RequireSignedCommits)
+
+	// Handle optional string attributes - use null if empty
+	if protection.ProtectedFilePatterns != "" {
+		data.ProtectedFilePatterns = types.StringValue(protection.ProtectedFilePatterns)
+	} else {
+		data.ProtectedFilePatterns = types.StringNull()
+	}
+
+	if protection.UnprotectedFilePatterns != "" {
+		data.UnprotectedFilePatterns = types.StringValue(protection.UnprotectedFilePatterns)
+	} else {
+		data.UnprotectedFilePatterns = types.StringNull()
+	}
+
+	data.EnableMergeWhitelist = types.BoolValue(protection.EnableMergeWhitelist)
+	data.EnableApprovalsWhitelist = types.BoolValue(protection.EnableApprovalsWhitelist)
+
+	// Handle required_approvals - use null if zero
+	if protection.RequiredApprovals != 0 {
+		data.RequiredApprovals = types.Int64Value(protection.RequiredApprovals)
+	} else {
+		data.RequiredApprovals = types.Int64Null()
+	}
+
+	data.BlockOnRejectedReviews = types.BoolValue(protection.BlockOnRejectedReviews)
+	data.BlockOnOfficialReviewRequests = types.BoolValue(protection.BlockOnOfficialReviewRequests)
+	data.BlockOnOutdatedBranch = types.BoolValue(protection.BlockOnOutdatedBranch)
+	data.DismissStaleApprovals = types.BoolValue(protection.DismissStaleApprovals)
+	data.PushWhitelistUsernames = types.ListNull(types.StringType)
+	data.PushWhitelistTeams = types.ListNull(types.StringType)
+	data.StatusCheckContexts = types.ListNull(types.StringType)
+	data.MergeWhitelistUsernames = types.ListNull(types.StringType)
+	data.MergeWhitelistTeams = types.ListNull(types.StringType)
+	data.ApprovalsWhitelistUsernames = types.ListNull(types.StringType)
+	data.ApprovalsWhitelistTeams = types.ListNull(types.StringType)
+
+	if len(protection.PushWhitelistUsernames) > 0 {
+		elements := make([]attr.Value, len(protection.PushWhitelistUsernames))
+		for i, username := range protection.PushWhitelistUsernames {
+			elements[i] = types.StringValue(username)
+		}
+		data.PushWhitelistUsernames, _ = types.ListValue(types.StringType, elements)
```

Comment:
You're silently ignoring `ListValue()` errors here (and in other places). This means you're discarding diagnostic data, allowing invalid state to be written...